lib/active_merchant/billing/gateways/element.rb

```diff
@@ -0,0 +1,267 @@
+          xml.TerminalID "01"
+          xml.CardPresentCode "UseDefault"
+          xml.CardholderPresentCode "UseDefault"
+          xml.CardInputCode "UseDefault"
+          xml.CVVPresenceCode "UseDefault"
+          xml.TerminalCapabilityCode "UseDefault"
+          xml.TerminalEnvironmentCode "UseDefault"
+          xml.MotoECICode "UseDefault"
+        end
+      end
+
+      def add_credit_card(xml, payment)
+        xml.card do
+          xml.CardNumber payment.number
+          xml.ExpirationMonth format(payment.month, :two_digits)
+          xml.ExpirationYear format(payment.year, :two_digits)
+          xml.CardholderName payment.first_name + " " + payment.last_name
+          xml.CVV payment.verification_value
+        end
+      end
+
+      def add_address(xml, options)
+        if address = options[:billing_address] || options[:address]
+          xml.address do
+            xml.BillingAddress1 address[:address1] if address[:address1]
+            xml.BillingAddress2 address[:address2] if address[:address2]
+            xml.BillingCity address[:city] if address[:city]
+            xml.BillingState address[:state] if address[:state]
+            xml.BillingZipcode address[:zip] if address[:zip]
+            xml.BillingEmail address[:email] if address[:email]
+            xml.BillingPhone address[:phone_number] if address[:phone_number]
+          end
+        end
+      end
+
+      def parse(xml)
+        response = {}
+
+        doc = Nokogiri::XML(xml)
+        doc.remove_namespaces!
+        doc.root.xpath("//response/*").each do |node|
+          if (node.elements.empty?)
+            response[node.name.downcase] = node.text
+          else
+            node_name = node.name.downcase
+            response[node_name] = Hash.new
+
+            node.elements.each do |childnode|
+              response[node_name][childnode.name.downcase] = childnode.text
+            end
+          end
+        end
+
+        response
+      end
+
+      def commit(action, xml, amount)
+        url = (test? ? test_url : live_url)
+        response = parse(ssl_post(url, xml, headers(action)))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: "#{authorization_from(response)}|#{amount}",
+          avs_result: AVSResult.new(code: response["some_avs_response_key"]),
+          cvv_result: CVVResult.new(response["some_cvv_response_key"]),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def success_from(response)
+        response["expressresponsecode"] == "0"
+      end
+
+      def message_from(response)
+        response["expressresponsemessage"]
+      end
+
+      def authorization_from(response)
+        response["transaction"]["transactionid"]
+      end
+
+      def error_code_from(response)
+        unless success_from(response)
+          # TODO: lookup error code for this response
```

Comment:
Ahh, good catch. Actually not supposed to be filled in. I'm just going to :hocho: it.